calcom/cal.com
# chore: refactor app list into common component

## What does this PR do?

I refactored the App List component used in settings/conferencing and installed/[category], into one component file.

Fixes #7521 

## Type of change



- Chore (refactoring code, technical debt, workflow improvements)


## How should this be tested?

- [ x] Test A : open both app list in settings/my-account/conferencing and apps/installed/conferencing. check if the app list is rendered, try to delete and and set new default. Changes on one should reflect on the other. 

## Mandatory Tasks

- [ x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.
